File: NativeTablet/AppDock/app/src/main/java/com/appdockproject/appdock/Data/AppViewHolder.java
package com.appdockproject.appdock.Data;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.appdockproject.appdock.R;

import static android.os.Build.VERSION_CODES.M;

/**
 * Created by jangerhard on 02-May-17.
 */

public class AppViewHolder extends RecyclerView.ViewHolder {

    public ImageView logo;
    TextView appName;

    public void setAppName(String name){
        appName.setText(name);
    }
    public AppViewHolder(View itemView) {
        super(itemView);

        logo = (ImageView) itemView.findViewById(R.id.appLogo);
        appName = (TextView) itemView.findViewById(R.id.appName);

        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mClickListener.onItemClick(v, getAdapterPosition());

            }
        });
    }
    private AppViewHolder.ClickListener mClickListener;

    //Interface to send callbacks...
    public interface ClickListener{
        void onItemClick(View view, int position);
    }

    public void setOnClickListener(AppViewHolder.ClickListener clickListener){
        mClickListener = clickListener;
    }
}

File: NativeTablet/AppDock/app/src/main/java/com/appdockproject/appdock/Data/AppViewHolder.kt
package com.appdockproject.appdock.Data

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.appdockproject.appdock.R

import android.os.Build.VERSION_CODES.M

/**
 * Created by jangerhard on 02-May-17.
 */

class AppViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    var logo: ImageView = itemView.findViewById(R.id.appLogo) as ImageView
    var appName: TextView = itemView.findViewById(R.id.appName) as TextView

    fun setAppName(name: String) {
        appName.text = name
    }

    init {

        itemView.setOnClickListener { v -> mClickListener!!.onItemClick(v, adapterPosition) }
    }

    private var mClickListener: AppViewHolder.ClickListener? = null

    //Interface to send callbacks...
    interface ClickListener {
        fun onItemClick(view: View, position: Int)
    }

    fun setOnClickListener(clickListener: AppViewHolder.ClickListener) {
        mClickListener = clickListener
    }
}
